dogstatsd-client/src/lib.rs

```diff
@@ -133,31 +129,55 @@ impl Client {
         &self,
         actions: Vec<DogStatsDAction<'a, T, V>>,
     ) {
-        let client = &self.client;
-
-        for action in actions {
-            if let Err(err) = match action {
-                DogStatsDAction::Count(metric, value, tags) => {
-                    let metric_builder = client.count_with_tags(metric.as_ref(), value);
-                    do_send(metric_builder, tags)
-                }
-                DogStatsDAction::Distribution(metric, value, tags) => {
-                    do_send(client.distribution_with_tags(metric.as_ref(), value), tags)
-                }
-                DogStatsDAction::Gauge(metric, value, tags) => {
-                    do_send(client.gauge_with_tags(metric.as_ref(), value), tags)
-                }
-                DogStatsDAction::Histogram(metric, value, tags) => {
-                    do_send(client.histogram_with_tags(metric.as_ref(), value), tags)
-                }
-                DogStatsDAction::Set(metric, value, tags) => {
-                    do_send(client.set_with_tags(metric.as_ref(), value), tags)
+        let client_opt = match self.get_or_init_client() {
+            Ok(guard) => guard,
```

Comment:
Nitpick: variable name `guard` is a bit confusing since it's the client option now.

Reply:
ah yes, forgot to update this variable. I'll make a note to update it in another PR.